### PR TITLE
feat: scramble IBAN in history event notes

### PR DESCRIPTION
**Pull request by lukicenturi**
Closes #(issue_number)

## Checklist

- [ ] The PR modified the frontend, and updated the [user guide](https://github.com/rotki/rotki/blob/develop/docs/usage_guide.rst) to reflect the changes.

### CODE REVIEW

**Comment by yabirgb** [comment]
Mentioned in db but there is an issue with the numbers 
image

**Comment by lukicenturi** [comment]
> Mentioned in db but there is an issue with the numbers image

should be fixed now

**Comment by yabirgb** [comment]
It works now and also thje scramble is consistent so the same IBAN get's the same scrambled value 
image

**Review by yabirgb — APPROVED** [review]
functionality looks good

File: frontend/app/src/composables/history/events/notes.ts [hunk 1]
```diff
@@ -55,24 +56,56 @@ export function useHistoryEventNote() {
...
+              // Extract and split the BBAN part into individual groups by spaces
+              const bban = match[3].trim();
+              const bbanGroups = bban.split(/\s+/);
+
+              const scrambledBbanNumbers = bbanGroups.map(item => scrambleIdentifier(item, 1000, 9999));
+              const formatted = `XX${scrambledCheckDigit} ${scrambledBbanNumbers.join(' ')}`;
+
+              notesVal = notesVal.replace(new RegExp(iban, 'g'), formatted);
+            }
+          });
```
Comment (kelsos) [inline frontend/app/src/composables/history/events/notes.ts:105]: @lukicenturi do you think we can extract a function here and maybe also add a small test for the scrambling?

File: frontend/app/tests/unit/specs/composables/history/notes.spec.ts [hunk 2]
```diff
@@ -293,6 +295,26 @@ describe('composables::history/notes', () => {
     expect(formatted).toMatchObject(expected);
   });
 
+  it('scramble IBAN', () => {
```
Comment (kelsos) [inline frontend/app/tests/unit/specs/composables/history/notes.spec.ts:298]: Wonderful thank you :)

File: frontend/app/src/composables/history/events/notes.ts (hunk 1 above)
Comment (lukicenturi) [reply frontend/app/src/composables/history/events/notes.ts]: updated

**Pull request merged**